File: main/hw2/python/environment.py
#!/usr/bin/env python3
import os

print("Cache-Control: no-cache")
print("Content-type: text/html\n")

print("<!DOCTYPE html>")
print("<html><head><title>Environment Variables (Python)</title></head>")
print("<body><h1 align='center'>Environment Variables (Python)</h1><hr>")

# Loop through all environment variables
for key, value in sorted(os.environ.items()):
    print(f"<b>{key}:</b> {value}<br />")

# --- NAVIGATION ---
print("<hr>")
print("<div style='text-align:center;'>")
print("<a href='/index.html'>Home</a>")
print("</div>")

print("</body></html>")

File: main/hw2/ruby/environment.rb
#!/usr/bin/ruby
puts "Cache-Control: no-cache"
puts "Content-type: text/html\n\n"

puts "<!DOCTYPE html><html><head><title>Environment Variables (Ruby)</title></head>"
puts "<body><h1 align='center'>Environment Variables (Ruby)</h1><hr>"

# Loop through ENV and sort keys
ENV.keys.sort.each do |key|
  puts "<b>#{key}:</b> #{ENV[key]}<br />"
end

puts "<hr><div style='text-align:center;'><a href='/index.html'>Home</a></div>"
puts "</body></html>"